File: ReglaDescuentoConsigna/ReglaDeDescuento.cs
namespace ReglaDescuentoConsigna
{
    public class ReglaDeDescuento
    {
        private static readonly double PorcentajeMenor = 0.03d;
        private static readonly double PorcentajeIntermedio = 0.05d;
        private static readonly double PorcentajeMayor = 0.10d;

        public double Calcular(double total)
        {
            if (total <= 0)
            {
                throw new ArgumentException("El total debe ser mayor a 0");
            }

            if (total > 5000 && total <= 10000)
            {
                return total * PorcentajeMenor;
            }
            else if (total > 10000 && total <= 25000)
            {
                return total * PorcentajeIntermedio;
            }
            else if (total > 25000)
            {
                return total * PorcentajeMayor;
            }

            return 0;
        }
    }

}

File: tienda/src/main/java/reglaDescuento/ReglaDeDescuento.java
package reglaDescuento;

public class ReglaDeDescuento {

    private static final double porcentajeMenor = 0.03d;
    private static final double porcentajeIntermedio = 0.05d;
    private static final double porcentajeMayor = 0.10d;

    public double Calcular(double total){

        if(total <= 0){
            throw new IllegalArgumentException("El total debe ser mayor a 0");
        }
        if(total > 5000 && total <= 10000){
            return total * porcentajeMenor;
        }
        if(total > 10000 && total <= 25000){
            return total * porcentajeIntermedio;
        }
        else if(total > 25000){
            return total * porcentajeMayor;
        }
        return 0;
    }

}
